src/parser.cpp

```diff
@@ -322,14 +322,18 @@ namespace ada::parser {
               url.path = base_url->path;
               url.query = base_url->query;
             } else  {
-              // TODO: This is likely not correct. There is a difference between having an empty username
-              // and not having a username.
-              url.update_base_username(base_url->get_username());
-              // TODO: This is likely not correct. There is a difference between having an empty password
-              // and not having a password.
-              url.update_base_password(base_url->get_password());
-              url.set_hostname(base_url->get_hostname());
-  	          url.update_base_port(base_url->retrieve_base_port());
+              if(base_url->has_non_empty_username()) {
+                url.update_base_username(base_url->get_username());
+              }
+              if(base_url->has_non_empty_password()) {
+                url.update_base_password(base_url->get_password());
+              }
```

Comment:
I think we should merge these two, like in my pull request. Faster, and a lot easier to handle the case of `//`

Reply:
Merging them is faster, but your PR creates temporary std::strings...

Reply:
Yes, I intend to remove it before merging. Its wip at the moment